iceoryx_utils/CMakeLists.txt

```diff
@@ -197,6 +197,12 @@ set_target_properties(iceoryx_platform PROPERTIES
 
 target_compile_options(iceoryx_platform PRIVATE ${ICEORYX_WARNINGS})
 
+target_link_libraries(iceoryx_platform 
+    PUBLIC 
+    rt 
+    pthread
+)
```

Comment:
after some thinking, I still think `rt` and `pthread` belongs to `iceoryx_platform` but it must only be added when the target is Linux. Maybe we need a separate `CMakeLists.txt` for each `platform/[linux,qnx,...]` folder

Reply:
Yes the linker errors in windows and Mac shows that it needs only to be added for linux. But i do not see the benefit of having more cmake files in comparison to have some platform checks in  utils cmake.